envbuilder.go

```diff
@@ -338,25 +353,45 @@ func Run(ctx context.Context, options Options) error {
 			}
 		}()
 
-		if options.GitUsername != "" || options.GitPassword != "" {
+		if options.GitSsh {
+			if options.GitSshkey != "" {
+				privateKey, err := LoadPrivateKey(options.GitUsername, options.GitSshkey)
+				if err != nil {
+					logf(codersdk.LogLevelError, "Failed to load private ssh key for Git: %s", err)
+				} else {
+					options.RepoAuth = privateKey
+				}
+			} else {
+				logf(codersdk.LogLevelInfo, "GIT_SSH set to True, load default ssh key from path: %s", "/root/.ssh/id_ed25519")
```

Comment:
I'm not sure this default path makes sense. The key doesn't need to be ED25519 specifically. Maybe it's better to require a path to the key.